// @syncfusion/ej2-angular-charts/esm2020/src/accumulation-chart/accumulationchart-all.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AccumulationChartModule } from './accumulationchart.module';
import { PieSeries, FunnelSeries, PyramidSeries, AccumulationTooltip, AccumulationLegend, AccumulationSelection, AccumulationDataLabel, AccumulationAnnotation } from '@syncfusion/ej2-charts';
import * as i0 from "@angular/core";
export const PieSeriesService = { provide: 'ChartsPieSeries', useValue: PieSeries };
export const FunnelSeriesService = { provide: 'ChartsFunnelSeries', useValue: FunnelSeries };
export const PyramidSeriesService = { provide: 'ChartsPyramidSeries', useValue: PyramidSeries };
export const AccumulationTooltipService = { provide: 'ChartsAccumulationTooltip', useValue: AccumulationTooltip };
export const AccumulationLegendService = { provide: 'ChartsAccumulationLegend', useValue: AccumulationLegend };
export const AccumulationSelectionService = { provide: 'ChartsAccumulationSelection', useValue: AccumulationSelection };
export const AccumulationDataLabelService = { provide: 'ChartsAccumulationDataLabel', useValue: AccumulationDataLabel };
export const AccumulationAnnotationService = { provide: 'ChartsAccumulationAnnotation', useValue: AccumulationAnnotation };
/**
 * NgModule definition for the AccumulationChart component with providers.
 */
export class AccumulationChartAllModule {
}
AccumulationChartAllModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: AccumulationChartAllModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccumulationChartAllModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: AccumulationChartAllModule, imports: [CommonModule, AccumulationChartModule], exports: [AccumulationChartModule] });
AccumulationChartAllModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: AccumulationChartAllModule, providers: [
        PieSeriesService,
        FunnelSeriesService,
        PyramidSeriesService,
        AccumulationTooltipService,
        AccumulationLegendService,
        AccumulationSelectionService,
        AccumulationDataLabelService,
        AccumulationAnnotationService
    ], imports: [[CommonModule, AccumulationChartModule], AccumulationChartModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: AccumulationChartAllModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, AccumulationChartModule],
                    exports: [
                        AccumulationChartModule
                    ],
                    providers: [
                        PieSeriesService,
                        FunnelSeriesService,
                        PyramidSeriesService,
                        AccumulationTooltipService,
                        AccumulationLegendService,
                        AccumulationSelectionService,
                        AccumulationDataLabelService,
                        AccumulationAnnotationService
                    ]
                }]
        }] });
