// @syncfusion/ej2-angular-charts/esm2020/src/chart/chart-all.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ChartModule } from './chart.module';
import { LineSeries, ScatterSeries, ColumnSeries, SplineSeries, SplineAreaSeries, StripLine, AreaSeries, ScrollBar, StepLineSeries, StepAreaSeries, StackingColumnSeries, StackingLineSeries, StackingAreaSeries, StackingStepAreaSeries, BarSeries, StackingBarSeries, RangeColumnSeries, BubbleSeries, Tooltip, Crosshair, Category, DateTime, Logarithmic, Legend, Zoom, DataLabel, Selection, ChartAnnotation, HiloSeries, HiloOpenCloseSeries, WaterfallSeries, RangeAreaSeries, SplineRangeAreaSeries, CandleSeries, PolarSeries, RadarSeries, SmaIndicator, TmaIndicator, EmaIndicator, AccumulationDistributionIndicator, MacdIndicator, AtrIndicator, RsiIndicator, MomentumIndicator, StochasticIndicator, BollingerBands, BoxAndWhiskerSeries, HistogramSeries, ErrorBar, Trendlines, DateTimeCategory, MultiColoredLineSeries, MultiColoredAreaSeries, MultiLevelLabel, ParetoSeries, Export, DataEditing, Highlight } from '@syncfusion/ej2-charts';
import * as i0 from "@angular/core";
export const LineSeriesService = { provide: 'ChartsLineSeries', useValue: LineSeries };
export const ScatterSeriesService = { provide: 'ChartsScatterSeries', useValue: ScatterSeries };
export const ColumnSeriesService = { provide: 'ChartsColumnSeries', useValue: ColumnSeries };
export const SplineSeriesService = { provide: 'ChartsSplineSeries', useValue: SplineSeries };
export const SplineAreaSeriesService = { provide: 'ChartsSplineAreaSeries', useValue: SplineAreaSeries };
export const StripLineService = { provide: 'ChartsStripLine', useValue: StripLine };
export const AreaSeriesService = { provide: 'ChartsAreaSeries', useValue: AreaSeries };
export const ScrollBarService = { provide: 'ChartsScrollBar', useValue: ScrollBar };
export const StepLineSeriesService = { provide: 'ChartsStepLineSeries', useValue: StepLineSeries };
export const StepAreaSeriesService = { provide: 'ChartsStepAreaSeries', useValue: StepAreaSeries };
export const StackingColumnSeriesService = { provide: 'ChartsStackingColumnSeries', useValue: StackingColumnSeries };
export const StackingLineSeriesService = { provide: 'ChartsStackingLineSeries', useValue: StackingLineSeries };
export const StackingAreaSeriesService = { provide: 'ChartsStackingAreaSeries', useValue: StackingAreaSeries };
export const StackingStepAreaSeriesService = { provide: 'ChartsStackingStepAreaSeries', useValue: StackingStepAreaSeries };
export const BarSeriesService = { provide: 'ChartsBarSeries', useValue: BarSeries };
export const StackingBarSeriesService = { provide: 'ChartsStackingBarSeries', useValue: StackingBarSeries };
export const RangeColumnSeriesService = { provide: 'ChartsRangeColumnSeries', useValue: RangeColumnSeries };
export const BubbleSeriesService = { provide: 'ChartsBubbleSeries', useValue: BubbleSeries };
export const TooltipService = { provide: 'ChartsTooltip', useValue: Tooltip };
export const CrosshairService = { provide: 'ChartsCrosshair', useValue: Crosshair };
export const CategoryService = { provide: 'ChartsCategory', useValue: Category };
export const DateTimeService = { provide: 'ChartsDateTime', useValue: DateTime };
export const LogarithmicService = { provide: 'ChartsLogarithmic', useValue: Logarithmic };
export const LegendService = { provide: 'ChartsLegend', useValue: Legend };
export const ZoomService = { provide: 'ChartsZoom', useValue: Zoom };
export const DataLabelService = { provide: 'ChartsDataLabel', useValue: DataLabel };
export const SelectionService = { provide: 'ChartsSelection', useValue: Selection };
export const ChartAnnotationService = { provide: 'ChartsChartAnnotation', useValue: ChartAnnotation };
export const HiloSeriesService = { provide: 'ChartsHiloSeries', useValue: HiloSeries };
export const HiloOpenCloseSeriesService = { provide: 'ChartsHiloOpenCloseSeries', useValue: HiloOpenCloseSeries };
export const WaterfallSeriesService = { provide: 'ChartsWaterfallSeries', useValue: WaterfallSeries };
export const RangeAreaSeriesService = { provide: 'ChartsRangeAreaSeries', useValue: RangeAreaSeries };
export const SplineRangeAreaSeriesService = { provide: 'ChartsSplineRangeAreaSeries', useValue: SplineRangeAreaSeries };
export const CandleSeriesService = { provide: 'ChartsCandleSeries', useValue: CandleSeries };
export const PolarSeriesService = { provide: 'ChartsPolarSeries', useValue: PolarSeries };
export const RadarSeriesService = { provide: 'ChartsRadarSeries', useValue: RadarSeries };
export const SmaIndicatorService = { provide: 'ChartsSmaIndicator', useValue: SmaIndicator };
export const TmaIndicatorService = { provide: 'ChartsTmaIndicator', useValue: TmaIndicator };
export const EmaIndicatorService = { provide: 'ChartsEmaIndicator', useValue: EmaIndicator };
export const AccumulationDistributionIndicatorService = { provide: 'ChartsAccumulationDistributionIndicator', useValue: AccumulationDistributionIndicator };
export const MacdIndicatorService = { provide: 'ChartsMacdIndicator', useValue: MacdIndicator };
export const AtrIndicatorService = { provide: 'ChartsAtrIndicator', useValue: AtrIndicator };
export const RsiIndicatorService = { provide: 'ChartsRsiIndicator', useValue: RsiIndicator };
export const MomentumIndicatorService = { provide: 'ChartsMomentumIndicator', useValue: MomentumIndicator };
export const StochasticIndicatorService = { provide: 'ChartsStochasticIndicator', useValue: StochasticIndicator };
export const BollingerBandsService = { provide: 'ChartsBollingerBands', useValue: BollingerBands };
export const BoxAndWhiskerSeriesService = { provide: 'ChartsBoxAndWhiskerSeries', useValue: BoxAndWhiskerSeries };
export const HistogramSeriesService = { provide: 'ChartsHistogramSeries', useValue: HistogramSeries };
export const ErrorBarService = { provide: 'ChartsErrorBar', useValue: ErrorBar };
export const TrendlinesService = { provide: 'ChartsTrendlines', useValue: Trendlines };
export const DateTimeCategoryService = { provide: 'ChartsDateTimeCategory', useValue: DateTimeCategory };
export const MultiColoredLineSeriesService = { provide: 'ChartsMultiColoredLineSeries', useValue: MultiColoredLineSeries };
export const MultiColoredAreaSeriesService = { provide: 'ChartsMultiColoredAreaSeries', useValue: MultiColoredAreaSeries };
export const MultiLevelLabelService = { provide: 'ChartsMultiLevelLabel', useValue: MultiLevelLabel };
export const ParetoSeriesService = { provide: 'ChartsParetoSeries', useValue: ParetoSeries };
export const ExportService = { provide: 'ChartsExport', useValue: Export };
export const DataEditingService = { provide: 'ChartsDataEditing', useValue: DataEditing };
export const HighlightService = { provide: 'ChartsHighlight', useValue: Highlight };
/**
 * NgModule definition for the Chart component with providers.
 */
export class ChartAllModule {
}
ChartAllModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: ChartAllModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ChartAllModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: ChartAllModule, imports: [CommonModule, ChartModule], exports: [ChartModule] });
ChartAllModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: ChartAllModule, providers: [
        LineSeriesService,
        ScatterSeriesService,
        ColumnSeriesService,
        SplineSeriesService,
        SplineAreaSeriesService,
        StripLineService,
        AreaSeriesService,
        ScrollBarService,
        StepLineSeriesService,
        StepAreaSeriesService,
        StackingColumnSeriesService,
        StackingLineSeriesService,
        StackingAreaSeriesService,
        StackingStepAreaSeriesService,
        BarSeriesService,
        StackingBarSeriesService,
        RangeColumnSeriesService,
        BubbleSeriesService,
        TooltipService,
        CrosshairService,
        CategoryService,
        DateTimeService,
        LogarithmicService,
        LegendService,
        ZoomService,
        DataLabelService,
        SelectionService,
        ChartAnnotationService,
        HiloSeriesService,
        HiloOpenCloseSeriesService,
        WaterfallSeriesService,
        RangeAreaSeriesService,
        SplineRangeAreaSeriesService,
        CandleSeriesService,
        PolarSeriesService,
        RadarSeriesService,
        SmaIndicatorService,
        TmaIndicatorService,
        EmaIndicatorService,
        AccumulationDistributionIndicatorService,
        MacdIndicatorService,
        AtrIndicatorService,
        RsiIndicatorService,
        MomentumIndicatorService,
        StochasticIndicatorService,
        BollingerBandsService,
        BoxAndWhiskerSeriesService,
        HistogramSeriesService,
        ErrorBarService,
        TrendlinesService,
        DateTimeCategoryService,
        MultiColoredLineSeriesService,
        MultiColoredAreaSeriesService,
        MultiLevelLabelService,
        ParetoSeriesService,
        ExportService,
        DataEditingService,
        HighlightService
    ], imports: [[CommonModule, ChartModule], ChartModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: ChartAllModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ChartModule],
                    exports: [
                        ChartModule
                    ],
                    providers: [
                        LineSeriesService,
                        ScatterSeriesService,
                        ColumnSeriesService,
                        SplineSeriesService,
                        SplineAreaSeriesService,
                        StripLineService,
                        AreaSeriesService,
                        ScrollBarService,
                        StepLineSeriesService,
                        StepAreaSeriesService,
                        StackingColumnSeriesService,
                        StackingLineSeriesService,
                        StackingAreaSeriesService,
                        StackingStepAreaSeriesService,
                        BarSeriesService,
                        StackingBarSeriesService,
                        RangeColumnSeriesService,
                        BubbleSeriesService,
                        TooltipService,
                        CrosshairService,
                        CategoryService,
                        DateTimeService,
                        LogarithmicService,
                        LegendService,
                        ZoomService,
                        DataLabelService,
                        SelectionService,
                        ChartAnnotationService,
                        HiloSeriesService,
                        HiloOpenCloseSeriesService,
                        WaterfallSeriesService,
                        RangeAreaSeriesService,
                        SplineRangeAreaSeriesService,
                        CandleSeriesService,
                        PolarSeriesService,
                        RadarSeriesService,
                        SmaIndicatorService,
                        TmaIndicatorService,
                        EmaIndicatorService,
                        AccumulationDistributionIndicatorService,
                        MacdIndicatorService,
                        AtrIndicatorService,
                        RsiIndicatorService,
                        MomentumIndicatorService,
                        StochasticIndicatorService,
                        BollingerBandsService,
                        BoxAndWhiskerSeriesService,
                        HistogramSeriesService,
                        ErrorBarService,
                        TrendlinesService,
                        DateTimeCategoryService,
                        MultiColoredLineSeriesService,
                        MultiColoredAreaSeriesService,
                        MultiLevelLabelService,
                        ParetoSeriesService,
                        ExportService,
                        DataEditingService,
                        HighlightService
                    ]
                }]
        }] });
