// @syncfusion/ej2-angular-treegrid/esm2020/src/treegrid/treegrid-all.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TreeGridModule } from './treegrid.module';
import { Filter, Page, Sort, Reorder, Toolbar, Aggregate, Resize, ColumnMenu, ExcelExport, PdfExport, CommandColumn, ContextMenu, Edit, Selection, VirtualScroll, DetailRow, RowDD, Freeze, ColumnChooser, Logger, InfiniteScroll } from '@syncfusion/ej2-treegrid';
import * as i0 from "@angular/core";
export const FilterService = { provide: 'TreeGridFilter', useValue: Filter };
export const PageService = { provide: 'TreeGridPage', useValue: Page };
export const SortService = { provide: 'TreeGridSort', useValue: Sort };
export const ReorderService = { provide: 'TreeGridReorder', useValue: Reorder };
export const ToolbarService = { provide: 'TreeGridToolbar', useValue: Toolbar };
export const AggregateService = { provide: 'TreeGridAggregate', useValue: Aggregate };
export const ResizeService = { provide: 'TreeGridResize', useValue: Resize };
export const ColumnMenuService = { provide: 'TreeGridColumnMenu', useValue: ColumnMenu };
export const ExcelExportService = { provide: 'TreeGridExcelExport', useValue: ExcelExport };
export const PdfExportService = { provide: 'TreeGridPdfExport', useValue: PdfExport };
export const CommandColumnService = { provide: 'TreeGridCommandColumn', useValue: CommandColumn };
export const ContextMenuService = { provide: 'TreeGridContextMenu', useValue: ContextMenu };
export const EditService = { provide: 'TreeGridEdit', useValue: Edit };
export const SelectionService = { provide: 'TreeGridSelection', useValue: Selection };
export const VirtualScrollService = { provide: 'TreeGridVirtualScroll', useValue: VirtualScroll };
export const DetailRowService = { provide: 'TreeGridDetailRow', useValue: DetailRow };
export const RowDDService = { provide: 'TreeGridRowDD', useValue: RowDD };
export const FreezeService = { provide: 'TreeGridFreeze', useValue: Freeze };
export const ColumnChooserService = { provide: 'TreeGridColumnChooser', useValue: ColumnChooser };
export const LoggerService = { provide: 'TreeGridLogger', useValue: Logger };
export const InfiniteScrollService = { provide: 'TreeGridInfiniteScroll', useValue: InfiniteScroll };
/**
 * NgModule definition for the TreeGrid component with providers.
 */
export class TreeGridAllModule {
}
TreeGridAllModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: TreeGridAllModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TreeGridAllModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: TreeGridAllModule, imports: [CommonModule, TreeGridModule], exports: [TreeGridModule] });
TreeGridAllModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: TreeGridAllModule, providers: [
        FilterService,
        PageService,
        SortService,
        ReorderService,
        ToolbarService,
        AggregateService,
        ResizeService,
        ColumnMenuService,
        ExcelExportService,
        PdfExportService,
        CommandColumnService,
        ContextMenuService,
        EditService,
        SelectionService,
        VirtualScrollService,
        DetailRowService,
        RowDDService,
        FreezeService,
        ColumnChooserService,
        LoggerService,
        InfiniteScrollService
    ], imports: [[CommonModule, TreeGridModule], TreeGridModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: TreeGridAllModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, TreeGridModule],
                    exports: [
                        TreeGridModule
                    ],
                    providers: [
                        FilterService,
                        PageService,
                        SortService,
                        ReorderService,
                        ToolbarService,
                        AggregateService,
                        ResizeService,
                        ColumnMenuService,
                        ExcelExportService,
                        PdfExportService,
                        CommandColumnService,
                        ContextMenuService,
                        EditService,
                        SelectionService,
                        VirtualScrollService,
                        DetailRowService,
                        RowDDService,
                        FreezeService,
                        ColumnChooserService,
                        LoggerService,
                        InfiniteScrollService
                    ]
                }]
        }] });
