// @syncfusion/ej2-angular-buttons/esm2020/src/speed-dial/speeddial.component.mjs
import { __decorate } from "tslib";
import { Component, ChangeDetectionStrategy, ContentChild } from '@angular/core';
import { ComponentBase, ComponentMixins, setValue } from '@syncfusion/ej2-angular-base';
import { SpeedDial } from '@syncfusion/ej2-buttons';
import { Template } from '@syncfusion/ej2-angular-base';
import { SpeedDialItemsDirective } from './items.directive';
import * as i0 from "@angular/core";
export const inputs = ['animation', 'closeIconCss', 'content', 'cssClass', 'direction', 'disabled', 'enablePersistence', 'enableRtl', 'iconPosition', 'itemTemplate', 'items', 'locale', 'modal', 'mode', 'openIconCss', 'opensOnHover', 'popupTemplate', 'position', 'radialSettings', 'target', 'visible'];
export const outputs = ['beforeClose', 'beforeItemRender', 'beforeOpen', 'clicked', 'created', 'onClose', 'onOpen', 'visibleChange'];
export const twoWays = ['visible'];
/**
 * Represents the Angular SpeedDial Component.
 * ```html
 * <button ejs-speeddial content='Edit'></button>
 * ```
 */
let SpeedDialComponent = class SpeedDialComponent extends SpeedDial {
    constructor(ngEle, srenderer, viewContainerRef, injector) {
        super();
        this.ngEle = ngEle;
        this.srenderer = srenderer;
        this.viewContainerRef = viewContainerRef;
        this.injector = injector;
        this.tags = ['items'];
        this.element = this.ngEle.nativeElement;
        this.injectedModules = this.injectedModules || [];
        this.registerEvents(outputs);
        this.addTwoWay.call(this, twoWays);
        setValue('currentInstance', this, this.viewContainerRef);
        this.containerContext = new ComponentBase();
    }
    ngOnInit() {
        this.containerContext.ngOnInit(this);
    }
    ngAfterViewInit() {
        this.containerContext.ngAfterViewInit(this);
    }
    ngOnDestroy() {
        this.containerContext.ngOnDestroy(this);
    }
    ngAfterContentChecked() {
        this.tagObjects[0].instance = this.childItems;
        this.containerContext.ngAfterContentChecked(this);
    }
};
SpeedDialComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: SpeedDialComponent, deps: [{ token: i0.ElementRef }, { token: i0.Renderer2 }, { token: i0.ViewContainerRef }, { token: i0.Injector }], target: i0.ɵɵFactoryTarget.Component });
SpeedDialComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.3", type: SpeedDialComponent, selector: "[ejs-speeddial]", inputs: { animation: "animation", closeIconCss: "closeIconCss", content: "content", cssClass: "cssClass", direction: "direction", disabled: "disabled", enablePersistence: "enablePersistence", enableRtl: "enableRtl", iconPosition: "iconPosition", itemTemplate: "itemTemplate", items: "items", locale: "locale", modal: "modal", mode: "mode", openIconCss: "openIconCss", opensOnHover: "opensOnHover", popupTemplate: "popupTemplate", position: "position", radialSettings: "radialSettings", target: "target", visible: "visible" }, outputs: { beforeClose: "beforeClose", beforeItemRender: "beforeItemRender", beforeOpen: "beforeOpen", clicked: "clicked", created: "created", onClose: "onClose", onOpen: "onOpen", visibleChange: "visibleChange" }, queries: [{ propertyName: "itemTemplate", first: true, predicate: ["itemTemplate"], descendants: true }, { propertyName: "popupTemplate", first: true, predicate: ["popupTemplate"], descendants: true }, { propertyName: "childItems", first: true, predicate: SpeedDialItemsDirective, descendants: true }], usesInheritance: true, ngImport: i0, template: `<ng-content ></ng-content>`, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
__decorate([
    Template()
], SpeedDialComponent.prototype, "itemTemplate", void 0);
__decorate([
    Template()
], SpeedDialComponent.prototype, "popupTemplate", void 0);
SpeedDialComponent = __decorate([
    ComponentMixins([ComponentBase])
], SpeedDialComponent);
export { SpeedDialComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: SpeedDialComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '[ejs-speeddial]',
                    inputs: inputs,
                    outputs: outputs,
                    template: `<ng-content ></ng-content>`,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    queries: {
                        childItems: new ContentChild(SpeedDialItemsDirective)
                    }
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.Renderer2 }, { type: i0.ViewContainerRef }, { type: i0.Injector }]; }, propDecorators: { itemTemplate: [{
                type: ContentChild,
                args: ['itemTemplate']
            }], popupTemplate: [{
                type: ContentChild,
                args: ['popupTemplate']
            }] } });
